pkg/controller/subscription/rhmiConfigs/main.go

```diff
@@ -164,18 +170,24 @@ func IsUpgradeServiceAffecting(csv *olmv1alpha1.ClusterServiceVersion) bool {
 }
 
 func ApproveUpgrade(ctx context.Context, client k8sclient.Client, installPlan *olmv1alpha1.InstallPlan, eventRecorder record.EventRecorder) error {
+	if installPlan.Spec.Approved == true {
+		return nil
+	}
 	if installPlan.Status.Phase == olmv1alpha1.InstallPlanPhaseInstalling {
 		return nil
 	}
 
 	eventRecorder.Eventf(installPlan, "Normal", integreatlyv1alpha1.EventUpgradeApproved,
 		"Approving %s install plan: %s", installPlan.Name, installPlan.Spec.ClusterServiceVersionNames[0])
 
-	installPlan.Spec.Approved = true
-	err := client.Update(ctx, installPlan)
+	_, err := controllerutil.CreateOrUpdate(ctx, client, installPlan, func() error {
```

Comment:
A lot of stuff modifies this CR at the same time, and it basically guaranteed an "object has been modified" error, doing the update this way stops that happening.

Reply:
Agreed